### PR TITLE
Waitlist table

### PR DESCRIPTION
**Pull request by mre**
Whenever we receive new credentials through `connect`, we store them in an encrypted table called the waitlist.

### CODE REVIEW

File: api/src/db/AuthRepository.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+        // Store the connectType in the database and get the session_id
+        await this.pool.query(sessionInsertStmt, [
+            'connectType',
+            connectType,
+            false,
+            sessionUUID,
+        ])
+
+        const insertStmt = `INSERT INTO podcastConnectWaitlist (
```
Comment (woolfg) [inline api/src/db/AuthRepository.ts:57]: no need to setup the prepared statement again, you can reuse the previous one.

File: api/src/db/AuthRepository.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            env_name,
+            env_value,
+            value_encrypted,
+            session_id
+            ) VALUES
+            (?,?,?,?)`
+
+        return await Promise.all(
```
Comment (woolfg) [inline api/src/db/AuthRepository.ts:65]: we should think about defining somehow to not encrypt a value, e.g. webstationid as it simplifies debugging if those non critical values are stored as plain text in the db

**Review by woolfg — CHANGES_REQUESTED** [review]
approach looks good

**Comment by mre** [comment]
Tested locally and works as expected. Needed to add `localhost:3001` to the list of allowed hosts for CORS.

**Pull request merged**